### PR TITLE
TNT-43446 Fix security vulnerabilities and bump dependencies

### PR DESCRIPTION
**Pull request by XDex**
## Description

Fix security vulnerabilities and bump dependencies

## Related Issue

https://jira.corp.adobe.com/browse/TNT-43446

## How Has This Been Tested?

Tests are passing

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: packages/target-decisioning-engine/src/index.spec.js [hunk 1]
```diff
@@ -77,7 +78,7 @@ describe("TargetDecisioningEngine", () => {
     expect(typeof decisioning.getOffers).toBe("function");
   });
 
-  // eslint-disable-next-line jest/no-test-callback
+  // eslint-disable-next-line jest/no-done-callback
```
Comment (jasonwaters) [inline packages/target-decisioning-engine/src/index.spec.js:81]: hmmm, 🤔  we use done callbacks pretty often in tests.  better to disable this rule in the global config as an override so it is permitted.

File: packages/target-decisioning-engine/src/postProcessors.js [hunk 2]
```diff
@@ -255,15 +255,13 @@ export function createResponseTokensPostProcessor(
...
-    const options = mboxResponse.options.map(option => {
-      return {
-        ...option,
-        responseTokens: {
-          ...responseTokensFromMeta,
-          ...responseTokens
-        }
-      };
-    });
+    const options = mboxResponse.options.map(option => ({
```
Comment (jasonwaters) [inline packages/target-decisioning-engine/src/postProcessors.js:258]: it's a matter of opinion, but i liked it better before

**Review by jasonwaters — APPROVED** [review]
I would like to push back a bit on PRs like this one where the entirety of the work is to update dependencies rather than add features or fix legitimate bugs.  It's true, we just saw issues with log4j vulnerabilities that caused us and every other big companies to frantically update the version of that dependency in their projects.  SO there are legitimately good reasons to keep dependencies updated, but not every dependency needs to be updated aggressively in my opinion.  In fact, doing so introduces risk and potential instability to the project.  Those updates add new code (that we didn't write) to the project and can potentially have new bugs and new vulnerabilities.  In general, I prefer to follow a "if it ain't broke, don't fix it" philosophy. 

Further reading on the topic:

- https://kevin.burke.dev/kevin/dont-update-dependencies/
- https://blog.kronis.dev/articles/never-update-anything

File: packages/target-decisioning-engine/src/index.spec.js (hunk 1 above)
Comment (XDex) [reply packages/target-decisioning-engine/src/index.spec.js:81]: Done - turned it off globally in `eslintrc.js`

File: packages/target-decisioning-engine/src/postProcessors.js (hunk 2 above)
Comment (XDex) [reply packages/target-decisioning-engine/src/postProcessors.js:258]: I don't have a strong opinion on this, however it was reported as an issue after ESLint upgrade and was fixed this way automatically after running `eslint --fix`

**Comment by XDex** [comment]
@jasonwaters Thank you for the prompt review! I totally agree with your point on not upgrading dependencies as soon as new versions come out and being extra careful when upgrading runtime dependencies, i.e. upgrading one-by-one, going through changelogs for breaking changes and thoroughly testing each upgrade, etc. However, in this PR's case I think the upgrade is warranted due to multiple high-severity and moderate security vulnerabilities in dev dependencies (the blogposts that you linked mention security upgrades as legitimate reasons for upgrades), which is still [a significant risk](https://security.stackexchange.com/a/256297) and it also doesn't look good when customers installing the NodeJS SDK are getting NPM audit warnings each time they install it..

**Pull request merged**